lib/ohai/plugins/linux/network.rb

```diff
@@ -171,6 +171,36 @@ def ethernet_layer_one(iface)
     iface
   end
 
+  # determine ring parameters for the interface using ethtool
+  def ethernet_ring_parameters(iface)
+    return iface unless ethtool_binary = find_ethtool_binary
+    iface.each_key do |tmp_int|
+      next unless iface[tmp_int][:encapsulation] == "Ethernet"
+      so = shell_out("#{ethtool_binary} -g #{tmp_int}")
+      Ohai::Log.debug("Parsing ethtool output: #{so.stdout}")
+      # Nasty, brittle regex to capture the ethtool output
```

Comment:
I'm open to suggestions. The main thing that makes this brittle to me is the reliance on perfect output matching, which means that any change in ethtool formatting of this would break it.